src/parser/alerts.ts

```diff
@@ -0,0 +1,130 @@
+ * Licensed under the MIT License: https://opensource.org/licenses/MIT
+ *
+ * Modifications Copyright (c) 2025 Tuure Piitulainen <https://github.com/tuurep>
+ */
+
+import MarkdownIt from 'markdown-it';
+import { config, configBaseDir } from '../config.js';
+import octicons from '@primer/octicons';
+import { existsSync, readFileSync } from 'fs';
+import { homedir } from 'os';
+import path from 'path';
+
+const githubAlertsIcons: Record<string, string> = {
+    note: 'info',
+    tip: 'light-bulb',
+    important: 'report',
+    warning: 'alert',
+    caution: 'stop',
+};
+const mergedIcons = {
+    ...githubAlertsIcons,
+    ...config.alertsOptions?.icons,
+};
+
+const fallbackIconOpt = config.alertsOptions?.fallbackIcon ?? mergedIcons['note'];
+const fallbackIcon = resolveIcon(fallbackIconOpt);
+const resolvedIcons: Record<string, string> = {};
+const titles = config.alertsOptions?.titles ?? {};
+
+for (const marker in mergedIcons) {
+    resolvedIcons[marker] = resolveIcon(mergedIcons[marker]);
+}
+
+function warnAndFallback(message: string): string {
+    return `<script>console.warn("${message}");</script>${fallbackIcon}`;
+}
+
+// Resolve option from alertsOptions.icons into raw svg tag
+function resolveIcon(iconOpt: string): string {
+    // Case 1: already a raw svg tag
+    if (iconOpt.startsWith('<svg')) {
+        return iconOpt;
+    }
+
+    // Case 2: svg file path
+    const prefix = ['/', './', '../', '~/'].find((p) => iconOpt.startsWith(p));
+    if (prefix && iconOpt.endsWith('.svg')) {
+        let iconPath = iconOpt;
+
+        if (prefix === '~/') {
+            iconPath = path.join(homedir(), iconPath.slice(2));
+        } else if (prefix === './' || prefix === '../') {
+            if (!configBaseDir) {
+                return warnAndFallback(`configBaseDir not set for relative icon path: ${iconPath}`);
+            }
+            iconPath = path.join(configBaseDir, iconPath);
+        }
+
+        if (!existsSync(iconPath)) {
+            return warnAndFallback(`Icon file not found: ${iconPath}`);
+        }
+        return readFileSync(iconPath).toString();
+    }
+
+    // Case 3: octicon name (in kebab-case) <https://primer.style/octicons>
+    const octiconName = iconOpt as keyof typeof octicons;
+    const octicon = octicons[octiconName]?.toSVG();
+
+    if (!octicon) {
+        return warnAndFallback(`Not a known octicon name: ${iconOpt}`);
+    }
+    return octicon;
+}
+
+function capitalize(str: string): string {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+}
+
+const MarkdownItAlerts = (md: MarkdownIt) => {
+    // Allow multi word alphanumeric markers (includes underscore)
+    // Additionally allow dashes in marker
+    const markerRE = '[\\w- ]+';
+
+    // Match marker case insensitively
+    // Note: config icons and titles keys must be fully lowercase
+    const RE = new RegExp(`^\\\\?\\[\\!(${markerRE})\\]([^\\n\\r]*)`, 'i');
```

Comment:
This can be simplified a lot if we use [Javascript's literal regex syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions#creating_a_regular_expression). But let's first talk about what the expression looks like as it is:

```
^\\?\[!([\w- ]+)\]([^\n\r]*)
```

What I find strange about this is that it looks for an optional backslash in the beginning, i.e. you can write `\[!note] title`, is that part of the spec? If not or we don't know I'd drop it.

Dropping it & using the literal syntax we arrive to

```suggestion
    // Match marker case-insensitively
    // Note: config icons and titles keys must be fully lowercase
    const RE = /^\[!([\w\- ]+)\]([^\n\r]*)/i;
```

Beyond this I also think the variable name `RE` is very unusual haha so some camelCased proper name would be better.

Reply:
Note: this part comes straight from original plugin and I didn't look at it, but yes let's make it as good as possible

Reply:
> What I find strange about this is that it looks for an optional backslash in the beginning, i.e. you can write `\[!note]` title, is that part of the spec? If not or we don't know I'd drop it.

Yes I don't understand the point to this, let's try removing it